Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx

```diff
@@ -873,9 +963,285 @@ struct HFTrackIndexSkimsCreator {
+          continue; // should go to the filter, but since it is a dynamic column, I cannot use it there
+        }
+
+        // V0 cosPointingAngle selection
+        if (v0.v0cosPA(collision.posX(), collision.posY(), collision.posZ()) < d_cosPAV0) {
+          MY_DEBUG_MSG(isK0SfromLc, LOG(INFO) << "K0S with daughters " << labelPos << " and " << labelNeg << ": rejected due to cosPA --> " << v0.v0cosPA(collision.posX(), collision.posY(), collision.posZ()) << " (cut " << d_cosPAV0 << ")");
+          continue;
+        }
+        MY_DEBUG_MSG(isK0SfromLc, LOG(INFO) << "KEPT! K0S from Lc with daughters " << labelPos << " and " << labelNeg);
+
+        auto posTrackParCov = getTrackParCov(posTrack);
+        posTrackParCov.propagateTo(v0.posX(), d_bZ); // propagate the track to the X closest to the V0 vertex
+        auto negTrackParCov = getTrackParCov(negTrack);
+        negTrackParCov.propagateTo(v0.negX(), d_bZ); // propagate the track to the X closest to the V0 vertex
+        std::array<float, 3> pVecV0 = {0., 0., 0.};
+        std::array<float, 3> pVecBach = {0., 0., 0.};
+
+        const std::array<float, 3> vertexV0 = {v0.x(), v0.y(), v0.z()};
+        const std::array<float, 3> momentumV0 = {v0.px(), v0.py(), v0.pz()};
+
+        // we build the neutral track to then build the cascade
+        auto trackV0 = o2::dataformats::V0(vertexV0, momentumV0, {0, 0, 0, 0, 0, 0}, posTrackParCov, negTrackParCov, {0, 0}, {0, 0}); // build the V0 track
+
+        // now we find the DCA between the V0 and the bachelor, for the cascade
+        int nCand2 = fitter.process(trackV0, bachTrack);
+        MY_DEBUG_MSG(isK0SfromLc && isProtonFromLc, LOG(INFO) << "Fitter result = " << nCand2 << " proton = " << protonLabel << " and K0S pos " << labelPos << " and neg " << labelNeg);
+        MY_DEBUG_MSG(isLc, LOG(INFO) << "Fitter result for true Lc = " << nCand2);
+        if (nCand2 == 0) {
+          continue;
+        }
+        std::array<float, 3> pVecCandCasc = {0., 0., 0.};
+        fitter.propagateTracksToVertex();        // propagate the bach and V0 to the Lc vertex
+        fitter.getTrack(0).getPxPyPzGlo(pVecV0); // take the momentum at the Lc vertex
+        fitter.getTrack(1).getPxPyPzGlo(pVecBach);
+
+        pVecCandCasc = array{pVecBach[0] + pVecV0[0],
+                             pVecBach[1] + pVecV0[1],
+                             pVecBach[2] + pVecV0[2]};
+
+        // cascade candidate pT cut
+        if (RecoDecay::Pt(pVecCandCasc) < d_cutCascPtCandMin) {
+          MY_DEBUG_MSG(isK0SfromLc && isProtonFromLc, LOG(INFO) << "True Lc from proton " << protonLabel << " and K0S pos " << labelPos << " and neg " << labelNeg << " rejected due to pt cut: " << RecoDecay::Pt(pVecCandCasc) << " (cut " << d_cutCascPtCandMin << ")");
+          continue;
+        }
+
+        // invariant mass
+        // calculate invariant masses
+        auto arrMom = array{pVecBach, pVecV0};
+        mass2K0sP = RecoDecay::M(arrMom, array{massP, massK0s});
```

Comment:
I already commented previously. Why don't you apply the pT and mass cuts before doing the CPU-expensive `fitter.process`?

Reply:
Now changed (I thought I commented already, I don't know why it is still here... ).